keanu-project/src/main/java/io/improbable/keanu/tensor/INDArrayShim.java

```diff
@@ -27,16 +37,37 @@
  */
 public class INDArrayShim {
 
+    /*
+     * We need to load ND4J in a separate thread as on load it sets the FTZ and DAZ flags in the processor for the
+     * thread that does the load.  This causes issues with Apache Math that makes use of Sub-normal values (in
+     * particular to initialisation values for the BrentOptimizer).
+     *
+     * We have raised https://github.com/deeplearning4j/deeplearning4j/issues/6690 to address this
+     */
+    public static void startNewThreadForNd4j() {
+        Thread nd4jInitThread = new Thread(() -> Nd4j.create(1));
+        nd4jInitThread.start();
+        try {
+            nd4jInitThread.join();
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+    }
+
     public static INDArray muli(INDArray left, INDArray right) {
         if (Arrays.equals(left.shape(), right.shape())) {
             return left.muli(right);
+        } else if (left.length() == 1) {
+            return applyScalarTensorOperationWithPreservedShape(left, right, INDArray::muli);
+        } else if (right.length() == 1) {
+            return applyScalarTensorOperationWithPreservedShape(right, left, INDArray::muli);
         } else {
             return broadcastMultiply(left, right);
         }
     }
 
     private static INDArray broadcastMultiply(INDArray a, INDArray b) {
```

Comment:
The `broadcast*` methods could also be deduped using lambdas (provided it doesn't impact perf)

Reply:
done